test/unit/slave/test_cluster_slave.py

```diff
@@ -195,11 +196,28 @@ def test_async_setup_happy_path_invokes_correct_methods(self):
         slave.connect_to_master(self._FAKE_MASTER_URL)
         project_type_mock = self.patch('app.slave.cluster_slave.util.create_project_type').return_value
         slave._project_type = project_type_mock
-        slave._async_setup_build([], [])
+        slave._async_setup_build([], {}, 0)
 
-        project_type_mock.fetch_project.assert_called_once_with([], [])
+        project_type_mock.fetch_project.assert_called_once_with()
         self.assertTrue(project_type_mock.run_job_config_setup.called)
 
+    def test_setup_build_sets_base_executor_index(self):
+        slave = self._create_cluster_slave()
+        slave.setup_build(build_id=123, project_type_params={'type': 'Fake'}, num_executors_already_allocated=8)
+        self.assertEqual(8, slave._base_executor_index, 'Build setup should set _base_executor_index')
+
+    def test_execute_subjob_passes_base_executor_index_to_executor(self):
+        slave = self._create_cluster_slave()
+        slave._base_executor_index = 12
+        slave._master_api = Mock()
+        executor = Mock()
+        slave._idle_executors = Mock()
+
+        with patch.object(builtins, 'open', mock_open(read_data='asdf')):
+            slave._execute_subjob(1, 2, executor, '', [])
+
+        executor.execute_subjob.assert_called_with(1, 2, '', [], 12)
```

Comment:
Adding keyword args on this line causes the test to fail, assert_called_with doesn't correctly compare a call using keywords with a call not using keywords I guess?  I added keywords to all the other test calls I could though